pkg/controller/workload/kubernetes/resource/resource.go

```diff
@@ -503,8 +503,12 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 
 	cluster, err := r.connect(ctx, ar)
 	if err != nil {
-		// TODO(negz): If err indicates the KubernetesCluster no longer exists
-		// we should just delete the resource, or we'll be stuck forever.
+		// If we're being deleted and we can't connect to our scheduled cluster
+		// because it doesn't exist we assume the cluster was deleted.
+		if ar.GetDeletionTimestamp() != nil && kerrors.IsNotFound(errors.Cause(err)) {
```

Comment:
Should the deletion check be conducted outside the `r.connect`?

Reply:
Hrm yeah good point it probably fits better in there. Will move it.

Reply:
Actually I believe it has to be here, at least with the current design. `r.connect` returns `(syncdeleter, error)`, so it could check the timestamp and remove the finalizer but it wouldn't have any way to stop further processing of the resource other than to return an error, which seems inappropriate.